### PR TITLE
onetbb/2021.9.0: add patch for correctly building with emscripten

### PR DESCRIPTION
**Pull request by ovostrikov**
Add patch for proper emscripten support.
Patch taken from yet unmerged PR in upstream.

Fixes https://github.com/conan-io/conan-center-index/issues/17776

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/onetbb/all/conandata.yml [hunk 1]
```diff
@@ -15,6 +15,11 @@ sources:
     url: "https://github.com/oneapi-src/oneTBB/archive/v2021.3.0.tar.gz"
     sha256: "8f616561603695bbb83871875d2c6051ea28f8187dbe59299961369904d1d49e"
 patches:
+  "2021.9.0":
+    - patch_description: "fix building for emscripten"
+      patch_type: "portability"
+      patch_source: "https://github.com/oneapi-src/oneTBB/pull/1006.patch"
```
Comment (memsharded) [inline recipes/onetbb/all/conandata.yml:21]: I am asking in https://github.com/oneapi-src/oneTBB/pull/1006#issuecomment-1584894267 what is the status of this patch, it is not that trivial, so some insights from the original author could be useful.

**Review by memsharded — COMMENTED** [review]
Thanks very much for the contribution. I am pinging upstream to get feedback about the patch status.

File: recipes/onetbb/all/conanfile.py [hunk 2]
```diff
@@ -90,6 +90,8 @@ def generate(self):
             toolchain.variables["TBB_ENABLE_IPO"] = self.options.interprocedural_optimization
         if Version(self.version) >= "2021.6.0" and self.options.get_safe("tbbproxy"):
             toolchain.variables["TBBMALLOC_PROXY_BUILD"] = self.options.tbbproxy
+        if self.settings.os == "Emscripten":
+            toolchain.variables["TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH"] = True
```
Comment (ilya-lavrenov) [inline recipes/onetbb/all/conanfile.py:94]: I suppose when we build for emscripten, it's a cross-compilation.
And during cross-compilation, hwloc is not found. See https://github.com/oneapi-src/oneTBB/blob/master/cmake/hwloc_detection.cmake#L48-L51

**Comment by danimtb** [comment]
@ovostrikov there are some conflicts in this PR now (sorry for that). I was going to fix them myself but I am not sure about the recent changes in the recipe. Maybe the changes in this PR are not needed, so better that you check it. Thank you!

**Comment by ilya-lavrenov** [comment]
Next oneTBB version (presumably 2021.11.0) will contain this patch. Maybe we should wait for official upstream support?

**Comment by ovostrikov** [comment]
Yes, I think it's not worth patching since proper upsteam support will arrive soon-ish. Closing PR

**Pull request closed without merging**